src/hdmf/utils.py

```diff
@@ -55,11 +53,7 @@ def __type_okay(value, argtype, allow_none=False):
             return __is_bool(value)
         return argtype in [cls.__name__ for cls in value.__class__.__mro__]
     elif isinstance(argtype, type):
-        if argtype == six.text_type:
-            return isinstance(value, six.text_type) or isinstance(value, six.string_types)
-        elif argtype == str:
-            return isinstance(value, six.string_types)
```

Comment:
It looks like this change removes the case of strings alltogether (not just the six.text_type case). Is this intentional?

Reply:
if argtype == str, then new line 62 will run: `return isinstance(value, argtype)` which will catch the case of strings.